src/Analyser/TypeSpecifier.php

```diff
@@ -347,6 +347,56 @@ public function specifyTypesInCondition(
 					$rootExpr,
 				);
 			}
+			if (
+				$expr->left instanceof ClassConstFetch &&
+				$expr->left->class instanceof Expr &&
+				$expr->left->name instanceof Node\Identifier &&
+				$expr->right instanceof ClassConstFetch &&
+				$rightType instanceof GenericClassStringType &&
+				strtolower($expr->left->name->toString()) === 'class'
+			) {
+				return $this->create(
+					$expr->left->class,
+					$rightType->getGenericType(),
+					$context,
+					false,
+					$scope,
+					$rootExpr,
+				);
+			}
+			if (
+				$expr->left instanceof FuncCall
+				&& $expr->left->name instanceof Name
+				&& strtolower($expr->left->name->toString()) === 'get_class'
+				&& isset($expr->left->getArgs()[0])
+				&& $rightType instanceof ConstantStringType
+			) {
+				return $this->specifyTypesInCondition(
+					$scope,
+					new Instanceof_(
+						$expr->left->getArgs()[0]->value,
+						new Name($rightType->getValue()),
+					),
+					$context,
+					$rootExpr,
+				);
+			}
```

Comment:
fyi: I had similar stuff in https://github.com/phpstan/phpstan-src/pull/1616 - but it did not work out.